.github/workflows/build.yaml

```diff
@@ -32,12 +32,24 @@ jobs:
       id: go  # XXX: what is this for?
     - name: Install govendor
       run: go get -u github.com/kardianos/govendor
-    # This dominates the duration of the test. Can we get a smaller set that is
-    # enough to build embedded C parts?
-    - name: Get C dependencies
+    - name: Cache Debian dependencies
+      id: cache-deb-downloads
+      uses: actions/cache@v1
+      with:
+        path: /tmp/debian-packages/
+        key: dpkg-debian-{{ hashFiles('**/debian/control') }}
+    - name: Download Debian dependencies
+      if: steps.cache-deb-downloads.outputs.cache-hit != 'true'
+      run: |
+          sudo apt clean
+          sudo apt update
+          sudo apt build-dep -d -y ${{ github.workspace }}/src/github.com/snapcore/snapd
+          mkdir -p /tmp/debian-packages/
+          cp /var/cache/apt/archives/*.deb /tmp/debian-packages
+    - name: Install Debian dependencies
       run: |
-          sudo apt-get update
-          sudo apt-get build-dep -y ./src/github.com/snapcore/snapd
+          cd /tmp/debian-packages
+          sudo apt install --no-install-recommends -y ./*.deb
```

Comment:
Could we copy /tmp/debian-packages/* to the /var/cache/apt/archives dir here and then sudo sudo apt-get build-dep -y ./src/github.com/snapcore/snapd ? should not make a difference in pratchse but feels more logical to me.

Reply:
Yeah, let me try.

Reply:
Although, that would also require a matching index. Otherwise the image with a stale index won't be able to satisfy those dependencies.